### PR TITLE
Don't symlink assets directory into the build

### PR DESCRIPTION
**Pull request by Matt-Yorkley**
Related to: https://github.com/openfoodfoundation/openfoodnetwork/issues/5643

The result of this symlink was that precompiled assets were shared between builds in a way that they shouldn't have been, and other issues with assets (especially in Rails 4) were obfuscated; old versions of non-digest assets which technically should *not* be available, actually are available **in some cases, on some servers, in a flaky way**. :see_no_evil:

:warning: We shouldn't merge this until the other assets cleanup PRs in the main repo are all tested and merged.

Additional note: manually deploying master to a staging server with this PR makes it really obvious which things need to be fixed. Any assets that still need to be fixed should be clearly missing after deploying then clearing browser cache / using a private tab. 

#### Dev test notes

- Pull this PR
- Deploy a server with it
- Check assets (fonts, CSS, JS, images, icons, etc) are OK on the site and nothing is missing/broken. Maybe check an email as well, and look for any image/CSS issues.

### CODE REVIEW

**Comment by Matt-Yorkley** [comment]
Another note: I thought some instances (including the UK) were keeping their Terms and conditions PDFs under `/shared/assets` (which would mean PDF links were broken by this change), but it turns out those copies are no longer used. Instances with custom PDFs (UK and France) are linking to external storage instead now, and not using files in that directory.

**Comment by luisramos0** [comment]
I moved to in dev, let us know when ready for review/not draft.

**Comment by Matt-Yorkley** [comment]
After #5682 and #5701 are merged, I think we'll be ready for this change. We should probably not merge it before the rollout is finished though.

**Comment by luisramos0** [comment]
Let's make this a dev-test.
Do you mind adding some testing guide for the dev testing this in staging?

**Review by sauloperez — APPROVED** [review]
pretty good catch @Matt-Yorkley :clap:

**Comment by sauloperez** [comment]
I, [2020-08-27 17:29:17#30609]  INFO -- : Started POST "/api/product_images/1895" for 109.69.11.78 at 2020-08-27 17:29:17 +0200                                                                                                                                 
I, [2020-08-27 17:29:17#30609]  INFO -- : Command :: identify -format '%wx%h,%[exif:orientation]' '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]'                                                                                        
I, [2020-08-27 17:29:17#30609]  INFO -- : Command :: identify -format %m '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]'                                                                                                                 
I, [2020-08-27 17:29:17#30609]  INFO -- : Command :: identify -format %m '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]'                                                                                                                 
I, [2020-08-27 17:29:17#30609]  INFO -- : Command :: identify -format %m '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]'                                                                                                                 
I, [2020-08-27 17:29:17#30609]  INFO -- : Command :: convert -auto-orient '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]' -auto-orient -resize "48x48>" -strip -auto-orient -colorspace RGB '/tmp/Screenshot from 2020-08-26 08-33-562020
0827-30609-jovj5k20200827-30609-1u5152q'                                                                                                                                                                                                                        
I, [2020-08-27 17:29:17#30609]  INFO -- : Command :: file -b --mime '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k20200827-30609-1u5152q'                                                                                                       
I, [2020-08-27 17:29:17#30609]  INFO -- : Command :: identify -format '%wx%h,%[exif:orientation]' '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]'                                                                                        
I, [2020-08-27 17:29:17#30609]  INFO -- : Command :: identify -format %m '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]'                                                                                                                 
I, [2020-08-27 17:29:17#30609]  INFO -- : Command :: identify -format %m '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]'                                                                                                                 
I, [2020-08-27 17:29:17#30609]  INFO -- : Command :: identify -format %m '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]'                                                                                                                 
I, [2020-08-27 17:29:17#30609]  INFO -- : Command :: convert -auto-orient '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]' -auto-orient -resize "100x100>" -strip -auto-orient -colorspace RGB '/tmp/Screenshot from 2020-08-26 08-33-5620
200827-30609-jovj5k20200827-30609-1qyhhxg'               
I, [2020-08-27 17:29:17#30609]  INFO -- : Command :: file -b --mime '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k20200827-30609-1qyhhxg'
I, [2020-08-27 17:29:17#30609]  INFO -- : Command :: identify -format '%wx%h,%[exif:orientation]' '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]'
I, [2020-08-27 17:29:17#30609]  INFO -- : Command :: identify -format %m '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]'
I, [2020-08-27 17:29:17#30609]  INFO -- : Command :: identify -format %m '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]'
I, [2020-08-27 17:29:17#30609]  INFO -- : Command :: identify -format %m '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]'
I, [2020-08-27 17:29:17#30609]  INFO -- : Command :: convert -auto-orient '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]' -auto-orient -resize "240x240>" -strip -auto-orient -colorspace RGB '/tmp/Screenshot from 2020-08-26 08-33-5620
200827-30609-jovj5k20200827-30609-7bif28'                  
I, [2020-08-27 17:29:17#30609]  INFO -- : Command :: file -b --mime '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k20200827-30609-7bif28'
I, [2020-08-27 17:29:17#30609]  INFO -- : Command :: identify -format '%wx%h,%[exif:orientation]' '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]'
I, [2020-08-27 17:29:17#30609]  INFO -- : Command :: identify -format %m '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]'
I, [2020-08-27 17:29:18#30609]  INFO -- : Command :: identify -format %m '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]'
I, [2020-08-27 17:29:18#30609]  INFO -- : Command :: identify -format %m '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]'
I, [2020-08-27 17:29:18#30609]  INFO -- : Command :: convert -auto-orient '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]' -auto-orient -resize "600x600>" -strip -auto-orient -colorspace RGB '/tmp/Screenshot from 2020-08-26 08-33-5620
200827-30609-jovj5k20200827-30609-1lcy2d6'         
I, [2020-08-27 17:29:18#30609]  INFO -- : Command :: file -b --mime '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k20200827-30609-1lcy2d6'
I, [2020-08-27 17:29:18#30609]  INFO -- : Command :: identify -format '%wx%h,%[exif:orientation]' '/tmp/Screenshot from 2020-08-26 08-33-5620200827-30609-jovj5k.png[0]'
I, [2020-08-27 17:29:18#30609]  INFO -- : [paperclip] Saving attachments.
I, [2020-08-27 17:29:18#30612]  INFO -- : Started GET "/spree/products/113/mini/Screenshot_from_2020-08-26_08-33-56.png" for 109.69.11.78 at 2020-08-27 17:29:18 +0200
I, [2020-08-27 17:29:18#30609]  INFO -- : Started GET "/spree/products/113/product/Screenshot_from_2020-08-26_08-33-56.png" for 109.69.11.78 at 2020-08-27 17:29:18 +0200
F, [2020-08-27 17:29:18#30609] FATAL -- : 
ActionController::RoutingError (No route matches [GET] "/spree/products/113/product/Screenshot_from_2020-08-26_08-33-56.png"):
  actionpack (4.0.13) lib/action_dispatch/middleware/debug_exceptions.rb:21:in `call'
  actionpack (4.0.13) lib/action_dispatch/middleware/show_exceptions.rb:30:in `call'

**Comment by Matt-Yorkley** [comment]
> I deployed from ofn-install's master and still, uploads show as a broken image.

So the issue is not related to this PR? Maybe a specific staging server image issue?

**Comment by RachL** [comment]
@Matt-Yorkley @sauloperez am I correct in thinking this should go back to the dev-test column?

**Comment by mkllnk** [comment]
I tested this on fr-staging and it's all looking good.

* I pulled this branch and merged master into it.
* I opened the page in a private browser and went through the checkout. All fine.
* I checked the default Terms and Conditions file and it's there (important for au).
* I logged into the backend and all looks fine there as well.
* I re-deployed again to test repeatability. :heavy_check_mark: 
* Oh, and then I send an email and it loaded a remote image correctly.

**Pull request merged**
